pkg/event/events.go

```diff
@@ -125,3 +125,37 @@ func NewBackgroundSuccessEvent(policy, rule string, source Source, r *unstructur
 
 	return events
 }
+
+func NewPolicyExceptionEvent(engineResponse *response.EngineResponse, ruleResp *response.RuleResponse) Info {
+	var reasonBuilder, messageBuilder strings.Builder
+	defer reasonBuilder.Reset()
+	defer messageBuilder.Reset()
```

Comment:
is this (resetting) necessary for local variables?

Reply:
I get this reference from the other two lines: 
- https://github.com/kyverno/kyverno/blob/main/pkg/event/events.go#L56
- https://github.com/kyverno/kyverno/blob/main/pkg/event/events.go#L76

I assume calling `sb.Reset` by defer would recycle the memory of the stringBuilder.

Not sure if I'm understanding correct..Are you suggesting that local variables would be automatically recycled through GC after the function is executed?

Reply:
Ok. We were seeing memory leaks in this area, so perhaps its done to help GC.

Reply:
Sounds good👍